src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,33 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
+                        // Don't render the whitespace between the start of the line and the razor comment.
+                        if (startOfLine && last.Type == HtmlSymbolType.WhiteSpace)
+                        {
+                            AddMarkerSymbolIfNecessary();
+                            Output(SpanKind.Markup);
```

Comment:
Talked offline; worth a comment reminding reader that `last` hasn't been accepted so this `Output()` handles what came _before_ the whitespace symbol.